code/game/machinery/_machines_base/stock_parts/network_lock.dm

```diff
@@ -39,18 +42,16 @@
 	if(!access_controller)
 		return
 
-	var/list/resulting_grants = list()
-	for(var/grant_data in grants)
-		var/datum/computer_file/data/grant_record/grant = access_controller.get_grant(grant_data)
-		if(!istype(grant))
-			continue // couldn't find.
-		resulting_grants |= uppertext("[D.network_id].[grant_data]")
-		if(legacy_compatibility_mode)
-			resulting_grants |= grant_data // This lets just grant_data, which is the access string for mapped machines, be used as an alternative.
-
-	if(!resulting_grants.len)
+	LAZYINITLIST(groups)
+	var/list/resulting_access = list()
+	for(var/group in groups)
+		if(!(group in access_controller.get_all_groups()))
+			groups -= groups // This group doesn't exist anymore - delete it.
+			continue
+		resulting_access |= "[group].[D.network_id]"
+	if(!resulting_access.len)
 		return
-	return list(resulting_grants) // List of lists is an OR type access configuration.
+	return OR_mode ? list(resulting_access) : resulting_access // List of lists is an OR type access configuration.
```

Comment:
Needs to be or by default to remain compatible with other logic. For the record, I hate that this thing doesn't work with full access format (nested lists), and have hated that from the time of its introduction.

Reply:
These now work with nested lists as requested.